plugin/helpers_bots.go

```diff
@@ -72,3 +113,67 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, botUserId string, options ...ShouldProcessMessageOption) bool {
+	messageProcessOptions := initialiseOptions(options)
+
+	if !messageProcessOptions.AllowSelf {
+		if post.UserId == botUserId {
+			return false
+		}
+	}
+
+	if !messageProcessOptions.AllowSystemMessages {
+		if post.IsSystemMessage() {
+			return false
+		}
+	}
+
+	if !contains(messageProcessOptions.FilterChannelIDs, post.ChannelId) {
+		channel, _ := p.API.GetChannel(post.ChannelId)
```

Comment:
The error should not be ignored. Please [`Wrap`](https://godoc.org/github.com/pkg/errors#Wrap) it and return it instead.

Reply:
Done